pkg/operator/controller/extension/virtualcluster.go

```diff
@@ -0,0 +1,195 @@
+				Resources: extension.Spec.Resources,
+				Deployment: &gardencorev1beta1.ControllerRegistrationDeployment{
+					Policy:       extension.Spec.Deployment.ExtensionDeployment.Policy,
+					SeedSelector: extension.Spec.Deployment.ExtensionDeployment.SeedSelector,
+					DeploymentRefs: []gardencorev1beta1.DeploymentRef{
+						{
+							Name: extension.Name,
+						},
+					},
+				},
+			}
+			return nil
+		})
+	return err
+}
+
+func (r *Reconciler) deleteVirtualClusterDeploymentResources(ctx context.Context, log logr.Logger, virtualClusterClient client.Client, extension *operatorv1alpha1.Extension) error {
+	log.Info("Deleting extension virtual resources")
+	var (
+		controllerDeployment = &gardencorev1.ControllerDeployment{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: extension.Name,
+			}}
+
+		controllerRegistration = &gardencorev1beta1.ControllerRegistration{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: extension.Name,
+			},
+		}
+	)
+
+	log.Info("Deleting ControllerRegistration and ControllerDeployment")
+	if err := kubernetesutils.DeleteObjects(ctx, virtualClusterClient, controllerDeployment, controllerRegistration); err != nil {
+		return err
+	}
+
+	log.Info("Waiting until ControllerRegistration is gone")
+	if err := kubernetesutils.WaitUntilResourceDeleted(ctx, virtualClusterClient, controllerRegistration, 5*time.Second); err != nil {
+		return err
+	}
+	r.Recorder.Event(extension, corev1.EventTypeNormal, "Deletion", "Successfully deleted ControllerRegistration")
+
+	log.Info("Waiting until ControllerDeployment is gone")
+	if err := kubernetesutils.WaitUntilResourceDeleted(ctx, virtualClusterClient, controllerDeployment, 5*time.Second); err != nil {
+		return err
+	}
+	r.Recorder.Event(extension, corev1.EventTypeNormal, "Deletion", "Successfully deleted ControllerDeployment")
+	return nil
+}
+
+func (r *Reconciler) reconcileAdmissionVirtualClusterResources(ctx context.Context, log logr.Logger, virtualClusterClientSet kubernetes.Interface, extension *operatorv1alpha1.Extension) error {
+	// return early if we do not have to make a deployment
+	if extension.Spec.Deployment == nil ||
+		extension.Spec.Deployment.AdmissionDeployment == nil ||
+		extension.Spec.Deployment.AdmissionDeployment.VirtualCluster == nil ||
+		extension.Spec.Deployment.AdmissionDeployment.VirtualCluster.Helm == nil {
+		return r.deleteAdmissionVirtualClusterResources(ctx, log, extension)
+	}
+
+	archive, err := r.HelmRegistry.Pull(ctx, extension.Spec.Deployment.AdmissionDeployment.VirtualCluster.Helm.OCIRepository)
+	if err != nil {
+		return fmt.Errorf("failed pulling Helm chart from OCI repository: %w", err)
+	}
+
+	gardenerValues := map[string]any{
+		"global": map[string]any{
+			"virtualGarden": map[string]any{
+				"enabled": true,
+				"user": map[string]any{
+					"name": fmt.Sprintf("system:serviceaccount:kube-system:%s", r.getVirtualClusterAccessSecret(extension).ServiceAccountName),
+				},
```

Comment:
```suggestion
```

Use `projectedKubeconfig` instead with the help of the generic token kubeconfig and a dedicated access token secret managed via GRM for this admission component

Reply:
That's missing but I don't think we can get rid of the "user" value, since it's required for RBAC ([example](https://github.com/gardener/gardener-extension-provider-aws/blob/7d7697417c245c4d55c25ef79d22c9ddc0e80ed8/charts/gardener-extension-admission-aws/charts/application/templates/rbac.yaml#L58)).

Reply:
Replaced by `ServiceAccount`